ui/dashboards/src/context/TemplateVariableProvider/hydrationUtils.test.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { VariableDefinition } from '@perses-dev/core';
+import { DEFAULT_ALL_VALUE } from '@perses-dev/plugin-system';
+import { hydrateTemplateVariableStates } from './hydrationUtils';
+
+describe('hydrateTemplateVariableStates', () => {
+  test('normalizes single "all" value in an array', () => {
```

Comment:
Only adding a test for the bug I'm fixing in this PR to get this fix out, but it may be a good idea to add more coverage here in the future.